### ISSUE TITLE
[Bug]: Galaxy Tab S9 Ultra keyboard cover touchpad operation issue

### ISSUE THREAD

**Issue by hansm629** [termux/termux-x11#661]
### Problem description

@knyipab

`Transform captured pointer movements` setting value
After selecting `Automatic (for touchpad)`

Turn the toggle switch on `Capture external pointer devices when possible`.
If you select `OFF`

The touchpad works very well in X11.
The pointer speed is consistent and all gestures are recognized well.

https://github.com/termux/termux-x11/assets/101167173/4c2b44d6-c2a4-4441-b9c9-0579ba73dc2d

But

Turn the toggle switch on `Capture external pointer devices when possible`.
If you select `ON`

The pointer speed becomes uncontrollably fast and gestures don't work as well as they do when turned off.

https://github.com/termux/termux-x11/assets/101167173/8fa8106d-1274-47d6-bf45-5f261ca37c2d

Even if you set `Capture external pointer devices when possible` to `ON`,
Is it possible to make the mouse pointer work the same way as when it is `OFF`?

### What steps will reproduce the bug?

Turn the toggle switch on `Capture external pointer devices when possible`.
If you select `ON`

**Comment by twaik** [comment]
Try lowering "Captured pointer speed factor" value.

**Comment by hansm629** [comment]
@twaik 
Even if you lower the value to 30%, nothing changes.

**Comment by twaik** [comment]
Does speed actually change or not?

**Comment by hansm629** [comment]
> Does speed actually change or not?

@twaik 
Upon closer inspection,
The pace has changed slightly.

However, fine control is still impossible.

**Comment by hansm629** [comment]
@twaik
`Capture external pointer devices when possible` switch
The `OFF` state is the most ideal usability.

The point speed was almost similar to the Android environment.


but
`Capture external pointer devices when possible` When the switch is turned `OFF`
It was inconvenient to use because the `top bar` and `bottom navigation` were recognized.

**Comment by twaik** [comment]
[app-arm64-v8a-debug.zip](https://github.com/user-attachments/files/16056053/app-arm64-v8a-debug.zip)
Try this apk and set speed factor to 1%.

**Comment by twaik** [comment]
@hansm629 ?

**Comment by hansm629** [comment]
@hansm629 
Test results from the new build
When the value is set to 5%, the pointer speed becomes normal.

However, there are the following problems:

- One-tap gestures are not recognized the first time (Gesture recognition seems to have slowed down as the pointer speed has slowed down.)
- Very slow point speed when using Bluetooth or USB mouse (not recognized even if adjusted in XFCE4 settings)

**Comment by twaik** [comment]
First tap always activates pointer capturing. It is android security restriction which can not be overridden.
The speed factor is shared between all captured devices, including BT and USB mice.

**Comment by hansm629** [comment]
@twaik
`Capture external pointer devices when possible`
When the switch is `ON`

Operates the same as the `OFF` state
Would it be difficult to make only the `top bar` and `bottom navigation` unrecognizable?

**Comment by twaik** [comment]
> When the switch is `ON`
> 
> Operates the same as the `OFF` state

I did not get it.

> Would it be difficult to make only the `top bar` and `bottom navigation` unrecognizable?

You mean disable notification and navigation bars without pointer capturing? Impossible.

**Comment by hansm629** [comment]
@twaik
I would like to know if the same symptom occurs on the Galaxy Tab S8 Plus keyboard cover touchpad, but I currently do not have it so I cannot test it.

First of all, in the current scenario
Only when the `Capture external pointer devices when possible` option is turned OFF
The Galaxy Tab S9 Ultra's keyboard cover touchpad seems to be working without a problem.


@knyipab 
Can you review the above symptoms?

**Comment by twaik** [comment]
> Only when the `Capture external pointer devices when possible` option is turned OFF The Galaxy Tab S9 Ultra's keyboard cover touchpad seems to be working without a problem.

In pointer capturing mode input events are delivered unmodified. In this mode application should decide how exactly input events should be handled. As far as I can see Android applies some kind of factor to input events, but unfortunately there is no public API to get it. That is the reason "Captured pointer speed factor" preference exists.

**Comment by twaik** [comment]
Fixed by a2e05cbcbf7162941fc88e206dcd098ab1902c39.

**Comment by knyipab** [comment]
> Very slow point speed when using Bluetooth or USB mouse (not recognized even if adjusted in XFCE4 settings)
> The speed factor is shared between all captured devices, including BT and USB mice.

A shared speed factor doesn't seem ideal when system-configured speeds can differ a lot across devices. I think the app could differentiate the input device type and apply different speed factor. Indeed, I implemented touchpad-specific speed factor in a branch of my fork here: https://github.com/knyipab/termux-x11/commit/234d605fd6d3d1d8d26511085449ad1027b13f0e. I did not make a PR cuz I thought it is too minor and make the preference page more clumsy. 

Now seeing the design of "additional keyboard" sub-page, it's perhaps a good idea to group "Transform captured pointer movements", "Captured pointer speed factor" and add device-specific (soft touchpad, hard touchpad, mouse and perhaps more?) speed factors into a sub-page of "Captured pointer".
